src/System.Text.RegularExpressions/src/System/Text/RegularExpressions/RegexInterpreter.cs

```diff
@@ -465,6 +461,13 @@ protected override void Go()
                     DumpState();
                 }
 #endif
+                if (advance >= 0)
```

Comment:
The DEBUG code isn't critical, but technically this block should come before the above if block, right?  Otherwise on iterations of the loop where calls to Advance(xyz) would have been made, the subsequent iteration would end up dumping the wrong state.

Reply:
Moved